deps/ReactantExtra/API.cpp

```
@@ -325,6 +325,74 @@ extern "C" PjRtDevice *ClientGetAddressableDevice(PjRtClient *client,
       client->LookupAddressableDevice(PjRtLocalDeviceId(device_id)));
 }
 
+extern "C" int64_t PjRtDeviceGetNumAllocs(PjRtDevice *device) {
+  auto stats = device->GetAllocatorStats();
+  if (!stats.ok())
+    return std::numeric_limits<int64_t>::min();
+  return stats->num_allocs;
+}
+extern "C" int64_t PjRtDeviceGetBytesInUse(PjRtDevice *device) {
+  auto stats = device->GetAllocatorStats();
+  if (!stats.ok())
+    return std::numeric_limits<int64_t>::min();
+  return stats->bytes_in_use;
+}
+extern "C" int64_t PjRtDeviceGetPeakBytesInUse(PjRtDevice *device) {
+  auto stats = device->GetAllocatorStats();
+  if (!stats.ok())
+    return std::numeric_limits<int64_t>::min();
+  return stats->peak_bytes_in_use;
+}
+extern "C" int64_t PjRtDeviceGetLargestAllocSize(PjRtDevice *device) {
+  auto stats = device->GetAllocatorStats();
+  if (!stats.ok())
+    return std::numeric_limits<int64_t>::min();
+  return stats->largest_alloc_size;
+}
+extern "C" int64_t PjRtDeviceGetBytesLimit(PjRtDevice *device) {
+  auto stats = device->GetAllocatorStats();
+  if (!stats.ok())
+    return std::numeric_limits<int64_t>::min();
+  return stats->bytes_limit.value_or(std::numeric_limits<int64_t>::min());
+}
+extern "C" int64_t PjRtDeviceGetBytesReserved(PjRtDevice *device) {
+  auto stats = device->GetAllocatorStats();
+  if (!stats.ok())
+    return std::numeric_limits<int64_t>::min();
+  return stats->bytes_reserved;
+}
+extern "C" int64_t PjRtDeviceGetPeakBytesReserved(PjRtDevice *device) {
+  auto stats = device->GetAllocatorStats();
+  if (!stats.ok())
+    return std::numeric_limits<int64_t>::min();
+  return stats->peak_bytes_reserved;
+}
+extern "C" int64_t PjRtDeviceGetBytesReservableLimit(PjRtDevice *device) {
+  auto stats = device->GetAllocatorStats();
+  if (!stats.ok())
```

Returning here is invalid since it expects an int return. Maybe throw an error (see myvalueorthrow)

---

I made it throw 👍